doc/en/user/source/styling/mbstyle/cookbook/polygons.rst

```diff
@@ -0,0 +1,165 @@
+            }
+          }
+      ]
+  }
+
+Details
+~~~~~~~
+
+There is one layer for this style, which is the simplest possible situation. Styling polygons is accomplished via the fill type (**line 7**). **Line 9** specifies dark blue (``'#000080'``) as the polygon's fill color.
+
+.. note::  The light-colored outlines around the polygons in the figure are artifacts of the renderer caused by the polygons being adjacent. There is no outline in this style.
+
+.. _mbstyle_cookbook_polygons_simplepolygonwithstroke:
+
+Simple polygon with stroke
+--------------------------
+
+This example adds a 1 pixel white outline to the :ref:`mbstyle_cookbook_polygons_simplepolygon` example.
+
+.. figure:: ../../sld/cookbook/images/polygon_simplepolygonwithstroke.png
+
+   Simple polygon with stroke
+
+Code
+~~~~
+
+:download:`Download the "Simple polygon with stroke" MBStyle <artifacts/mbstyle_polygon_simplepolygonwithstroke.json>`
+
+.. code-block:: json
+  :linenos:
+
+  {
+    "version": 8,
+    "name": "simple-polygon-outline",
+  	"layers":[
+      	  {
+            "id": "polygon-outline",
+            "type": "fill",
+            "paint": {
+              "fill-outline-color": "#FFFFFF",
+              "fill-color": "#000080"
+            }
+          }
+      ]
+  }
+
+Details
+~~~~~~~
+
+This example is similar to the :ref:`mbstyle_cookbook_polygons_simplepolygon` example above, with the addition of ``fill-outline`` paint parameter (**line 9**). **Line 9** also sets the color of stroke to white (``'#FFFFFF'``), the ``"fill-outline-color"`` can only be 1 pixel, a limitation of MBStyle.
+
+
+Transparent polygon
+-------------------
+
+This example builds on the :ref:`mbstyle_cookbook_polygons_simplepolygonwithstroke` example and makes the fill partially transparent by setting the opacity to 50%.
+
+.. figure:: ../../sld/cookbook/images/polygon_transparentpolygon.png
+
+   Transparent polygon
+
+Code
+~~~~
+
+:download:`Download the "Transparent polygon" MBStyle <artifacts/mbstyle_polygon_transparentpolygon.json>`
+
+.. code-block:: json
+  :linenos:
+
+  {
+    "version": 8,
+    "name": "simple-polygon-outline",
+  	"layers":[
+      	  {
+            "id": "polygon-outline",
+            "type": "fill",
+            "paint": {
+              "fill-outline-color": "#FFFFFF",
+              "fill-color": "#000080",
+              "fill-opacity": "0.5"
+            }
+          }
+      ]
+  }
+
+Details
+~~~~~~~
+
+This example is similar to the :ref:`mbstyle_cookbook_polygons_simplepolygonwithstroke` example, save for defining the fill's opacity in **line 11**. The value of 0.5 results in partially transparent fill that is 50% opaque. An opacity value of 1 would draw the fill as 100% opaque, while an opacity value of 0 would result in a completely transparent (0% opaque) fill. In this example, since the background is white, the dark blue looks lighter. Were the points imposed on a dark background, the resulting color would be darker.
```

Comment:
"Were the points imposed on a dark background" - should this be "fill" instead of points?